File: 0001-two-sum/0001-two-sum.js
var twoSum = function (nums, target) {
  let counter = {};
  for (let i = 0; i < nums.length; i++) {
    if (counter[nums[i]] !== undefined) {
      return [counter[nums[i]], i];
    }
    let complement = target - nums[i];
    counter[complement] = i;
  }

  return [];
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> count = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int diff = target - nums[i];
            if(count.containsKey(diff)){
                return new int[] {count.get(diff), i};
            }else{
                count.put(nums[i], i);
            }
        }
        return new int[]{};
    }
}